content/en/blog/2020/custom-istio-dns-name/index.md

```diff
@@ -0,0 +1,125 @@
+
+This `ServiceEntry` creates an entry in the Istio service-discovery registry for `httpbin.bar.global` with endpoints that point to a different cluster's gateway. Without any other configuration changes, this hostname, `httpbin.bar.global` is not routable. In the docs for setting up multi-cluster replicated control plane, however, there is a [section for configuring the DNS stubbing](/docs/setup/install/multicluster/gateways/#setup-dns) for your Kubernetes cluster. When we set up the DNS stubbing, we tell `kube-dns` to delegate queries for `.global` to the Istio coredns component.
+
+But what if you don't want to use `.global` as the suffix? What if you have to keep some backward compatibility and the root domain name needs to be something custom? Or what if you want to be able to support multiple root domain suffixes? In the open-source project [Service Mesh Hub](https://github.com/solo-io/service-mesh-hub) that I work on, we had this exact scenario. Service Mesh Hub is used to [automate multi-cluster Istio service mesh deployments](https://docs.solo.io/service-mesh-hub/latest/getting_started/), and routing between clusters is a use case our users have. Here's how to configure it in Istio.
+
+## Istio CoreDNS
+
+Istio's operator allows you to deploy the Istio `coredns` addon component.  For example, you can specify the following in the operator config:
+
+{{< text yaml >}}
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  name: istiocontrolplane-default
+  namespace: istio-operator
+spec:
+  addonComponents:
+    istiocoredns:
+      enabled: true
+{{< /text >}}
+
+When this deploys, we have the `istiocoredns` component that has `coredns` installed as well as the `istio-coredns-plugin` which is used to read `ServiceEntry`s and serve DNS queries for their hostname with IP addresses. The interesting bit is how it's configured out of the box. The default out of the box config is in the `coredns` configmap:
+
+{{< text bash >}}
+$ kubectl get cm coredns -n istio-system
+{{< /text >}}
+
+Looking into that configmap, we see the following configuration for Istio's `coredns`:
+
+{{< text json >}}
+.:53 {
+      errors
+      health
+
+      # Removed support for the proxy plugin: https://coredns.io/2019/03/03/coredns-1.4.0-release/
+      grpc global 127.0.0.1:8053
+      forward . /etc/resolv.conf {
+        except global
+      }
+
+      prometheus :9153
+      cache 30
+      reload
+    }
+{{< /text >}}
+
+Understanding how [coredns plugins work](https://coredns.io/plugins/grpc/) is helpful here. We see with the `grpc` statement that we're forwarding `global` domains to `127.0.0.1:8053`, and we're forwarding everything else to `/etc/resolv.conf`. The `istio-coredns-plugin` plugin is another container co-deployed with the `coredns` server and is listening on that `127.0.0.1:8053` port within the pod. If we want to change or add domains that can resolve based on what `ServiceEntry`s are created, we need to modify this file.
```

Comment:
change all references of we to first person.

Reply:
"we" is first person?

Reply:
I had meant third person. Been a long day - even at 2pm 👍 Please reference this style guideline for followup questions : https://developers.google.com/style/pronouns

Reply:
Yeah, best to avoid "we". In addition to the pronouns page that sdake pointed to, see also (https://developers.google.com/style/person).